src/Polly.Core/Simmy/Outcomes/OutcomeChaosPipelineBuilderExtensions.TResult.cs

```diff
@@ -0,0 +1,157 @@
+﻿using System.Diagnostics.CodeAnalysis;
+
+namespace Polly.Simmy.Outcomes;
+
+/// <summary>
+/// Extension methods for adding outcome to a <see cref="ResiliencePipelineBuilder"/>.
+/// </summary>
+public static partial class OutcomeChaosPipelineBuilderExtensions
+{
+    /// <summary>
+    /// Adds a fault chaos strategy to the builder.
+    /// </summary>
+    /// <typeparam name="TResult">The type of result the retry strategy handles.</typeparam>
+    /// <param name="builder">The builder instance.</param>
+    /// <param name="enabled">A value that indicates whether or not the chaos strategy is enabled for a given execution.</param>
+    /// <param name="injectionRate">The injection rate for a given execution, which the value should be between [0, 1].</param>
+    /// <param name="fault">The exception to inject.</param>
+    /// <returns>The builder instance with the retry strategy added.</returns>
+    public static ResiliencePipelineBuilder<TResult> AddChaosFault<TResult>(this ResiliencePipelineBuilder<TResult> builder, bool enabled, double injectionRate, Exception fault)
```

Comment:
@clpun since all the chaos options inherit from the `MonkeyStrategyOptions` class, both generators are available across all chaos policies, e.g: 

```C#
var options = new LatencyStrategyOptions
{
    EnabledGenerator = (_) => new ValueTask<bool>(true),
    InjectionRateGenerator = (_) => new ValueTask<double>(0.6),
    ...
};
```

Reply:
I see. Thank you for clarifying.